File: Core Java/collage-app/src/com/techlabs/collage/Professor.java
package com.techlabs.collage;

import com.techlabs.interfaces.ISalarayCalulator;

public class Professor extends Person implements ISalarayCalulator {

	private double salary;

	public Professor(int id, String address, String dob) {
		super(id, address, dob);
	}

	public double getSalary() {
		return salary;
	}

	@Override
	public String toString() {
		
		String details = "Professor Details.......\n";
		details += "Id : "+getId()+ "\n";
		details += "Address : " +getAddress()+ "\n";
		details += "DOB : " +getDob()+ "\n";
		details += "Salary : " +getSalary()+ "\n";
		return details ;
	}

	@Override
	public double calculateSalary(int days) {

		double oneDaySalary = 1000;
		salary = days * oneDaySalary;
		return salary;
	}

}

File: DOTNET/C#/CollageApp/CollageApp/Professor.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CollageApp
{
    class Professor : Person, ISalaryCalculator
    {
        private double salary;
        public Professor(int id, string address, string dob) : base(id, address, dob)
        {
        }

        public double Salary { get { return salary; } }

        public double CalculateSalary(int days)
        {
            double oneDaySalary = 1000;
            salary = days * oneDaySalary;
            return salary;
        }

        public override string ToString()
        {
            String details = "Professor Details....\n";
            details += "Id: " + Id + "\n";
            details += "Address: " + Address + "\n";
            details += "DOB: " + Dob + "\n";
            details += "Salary: " + Salary + "\n";
            return details;
        }
    }
}
